src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/RSACryptoServiceProvider.Unix.cs

```diff
@@ -0,0 +1,241 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Cryptography;
+using Internal.NativeCrypto;
+using System.IO;
+
+namespace System.Security.Cryptography
+{
+    public sealed partial class RSACryptoServiceProvider : RSA, ICspAsymmetricAlgorithm
+    {
+        private readonly RSA _impl;
+        private bool _publicOnly;
+
+        public RSACryptoServiceProvider()
+            : this(0, true) { }
+
+        public RSACryptoServiceProvider(int dwKeySize)
+            : this(dwKeySize, false) { }
+
+        public RSACryptoServiceProvider(int dwKeySize, CspParameters parameters)
+            : this(dwKeySize, false) { }
+
+        public RSACryptoServiceProvider(CspParameters parameters)
+            : this(0, true) { }
+
+        private RSACryptoServiceProvider(int keySize, bool useDefaultKeySize)
+        {
+            if (keySize < 0)
+                throw new ArgumentOutOfRangeException(nameof(keySize), SR.ArgumentOutOfRange_NeedNonNegNum);
+
+            // This class wraps RSA
+            _impl = RSA.Create();
+            _impl.KeySize = useDefaultKeySize ? 1024 : keySize;
```

Comment:
You'd need to be on a pretty recent master, it only merged in on Wednesday: https://github.com/dotnet/corefx/pull/17851